client/src/testFixtures/java/org/apache/cassandra/sidecar/client/SidecarClientTest.java

```diff
@@ -1673,6 +1677,42 @@ public void onError(Throwable throwable)
         assertThat(new String(baos.toByteArray(), StandardCharsets.UTF_8)).isEqualTo("Test Content");
     }
 
+    @Test
+    public void testReportSchemaSuccess()
+    {
+        MockResponse response = new MockResponse()
+                .setResponseCode(OK.code())
+                .setBody("");
+
+        enqueue(response);
+
+        SidecarInstance instance = instances.get(0);
+
+        CompletableFuture<Void> future = client.reportSchema(instance);
+
+        assertDoesNotThrow(() ->
+                future.whenComplete((ignored, throwable) -> assertNull(throwable))
+                      .join());
+    }
+
+    @Test
+    public void testReportSchemaFailure()
+    {
+        MockResponse response = new MockResponse()
+                .setResponseCode(INTERNAL_SERVER_ERROR.code())
+                .setBody("Message");
+
+        enqueue(response);
+
+        SidecarInstance instance = instances.get(0);
+
+        CompletableFuture<Void> future = client.reportSchema(instance);
+
+        assertThrows(CompletionException.class, () ->
+                future.whenComplete((ignored, throwable) -> assertNotNull(throwable))
+                      .join());
```

Comment:
Use assertJ and enrich the assertion

```suggestion
        SidecarInstance instance = instances.get(0);
        assertThatThrownBy(() -> client.reportSchema(instance).get())
        .isExactlyInstanceOf(ExecutionException.class)
        .hasCauseInstanceOf(RetriesExhaustedException.class)
        .hasMessageContaining("statusCode=500, statusMessage='Server Error'");
```

Reply:
Done.